test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/HubConnectionTests.cs

```diff
@@ -96,6 +96,41 @@ public HubConnectionTests(ServerFixture<Startup> serverFixture, ITestOutputHelpe
 
         [Theory]
         [MemberData(nameof(HubProtocolsAndTransportsAndHubPaths))]
+        public async Task CheckFixedMessage_HubConnectionOptions(IHubProtocol protocol, TransportType transportType, string path)
+        {
+            using (StartLog(out var loggerFactory))
+            {
+                var connection = new HubConnectionOptions
+                    {
+                        Url = new Uri(_serverFixture.BaseUrl + path),
+                        HubProtocol = protocol,
+                        TransportType = transportType,
+                        LoggerFactory = loggerFactory
+                    }.Create();
+
```

Comment:
I think I like the HubConnectionOptions pattern the best given the trade-offs of wrapping HubConnection (i.e. IHubConnection interface/virtual methods/copy of extension methods)